proposals/0360-opaque-result-types-with-availability.md

```diff
@@ -87,18 +87,139 @@ This is a dead-end for the library author although SE-0244 states that it should
+
+  ```swift
+  func test() -> some Shape {
+    if #available(macOS 100, *) { ✅
+      return Rectangle()
+    }
+
+    return self
+  }
+  ```
+  
+  The following is incorrect because `if #available` is preceded by a dynamic condition that returns:
+  
+  ```swift
+  func test() -> some Shape {
+    guard let x = <opt-value> else {
+      return ...
+    }
+    
+    if #available(macOS 100, *) { ❌
+      return Rectangle()
+    }
+
+    return self
+  }
+  ```
+
+  Similarly, the following is incorrect because `if #available` appears inside of a loop:
+
+  ```swift
+  func test() -> some Shape {
+    for ... {
+      if #available(macOS 100, *) { ❌
+        return Rectangle()
+      }
+    }
+    return self
+  }
+  ```
+
+2. All `return` statements outside of unconditional availability clauses must return the same type as each other, and this type must be as available as the containing function.
+
+  All `return` statements within a given unconditional availability clause must return the same type as each other, and this type must be as available as the `#available` condition of the clause. This type need not be the same type returned by any `return` statement outside of the clause.
+
+  There must be at least one `return` statement in the containing function.  If there are no `return` statements outside of unconditional availability clauses, then at least one of the return types within unconditional availability clauses must be as available as the containing function.
+
+  Dynamically, the return type of the containing function is:
+    - the return type of `return` statements in the first unconditional availability clause whose condition is dynamically satisfied, or if none are satisfied then
+    - the return type of `return` statements outside of all unconditional availability clauses, or if there are no such statements then
+    - the return type of `return` statements in the first unconditional availability clause that is as available as the containing function.
+
+  The following `test()` function is well-formed because `if` statement produces the same result in both of its branches and it's statically known that the `if #available` always terminates with a `return`
+  
+  ```swift
+  func test() -> some Shape {
+    if #available(macOS 100, *) {
+       if cond { ✅
+         return Rectangle(...)
+       } else {
+         return Rectangle(...)
+       }
+    }
+    return self
+  }
+  ```
+
+  But:
+
+  ```swift
+  func test() -> some Shape {
+    if #available(macOS 100, *) {
+       if cond { ❌
+         return Rectangle()
+       } else {
+         return Square()
+       }
+    }
+    return self
+  }
+  ```
+
+  is not going to be accepted by the compiler because return types are different: `Rectangle` vs. `Square`.
+
+3. `if #available` *cannot* appear in an `else` branch unless all preceding clauses are `if #available` conditions that meet all of the requirements listed in this section.
```

Comment:
These two examples belongs in the first set of examples, since they are about that first set of rules.  You don't need this lead-in sentence repeating the rule; just a sentence before each example describing why it satisfies or doesn't satisfy the rule is fine.